packages/amplify-graphql-generation-transformer/src/__tests__/amplify-graphql-generation-transformer.test.ts

```diff
@@ -0,0 +1,239 @@
+import { AuthTransformer } from '@aws-amplify/graphql-auth-transformer';
+import { IndexTransformer, PrimaryKeyTransformer } from '@aws-amplify/graphql-index-transformer';
+import { ModelTransformer } from '@aws-amplify/graphql-model-transformer';
+import { DDB_AMPLIFY_MANAGED_DATASOURCE_STRATEGY, validateModelSchema } from '@aws-amplify/graphql-transformer-core';
+import { AppSyncAuthConfiguration, ModelDataSourceStrategy } from '@aws-amplify/graphql-transformer-interfaces';
+import { DeploymentResources, testTransform } from '@aws-amplify/graphql-transformer-test-utils';
+import { parse } from 'graphql';
+import { GenerationTransformer } from '..';
+import { BelongsToTransformer, HasManyTransformer, HasOneTransformer } from '@aws-amplify/graphql-relational-transformer';
+
+const todoModel = `
+type Todo @model {
+  content: String
+  isDone: Boolean
+}`;
+
+test('generation route model list response type', () => {
+  const queryName = 'makeTodos';
+  const inputSchema = `
+    ${todoModel}
+
+    type Query {
+        ${queryName}(description: String!): [Todo]
+        @generation(
+          aiModel: "anthropic.claude-3-haiku-20240307-v1:0",
+          systemPrompt: "Make a list of todo items based on the description."
+        )
+        @auth(rules: [{ allow: public, provider: iam }])
+    }
+  `;
+
+  expect(() => transform(inputSchema)).toThrow(/Disallowed required field type/);
```

Comment:
The error message is much more descriptive, but it can fail on different fields (`createdAt` or `updatedAt`) here. Hence the generalized regex check. Here's the [error that's thrown](https://github.com/aws-amplify/amplify-category-api/pull/2820/files#diff-d8333a03a8ba937ab9767ac962175751e0e8583497f8d6152a007090cf3bfab6R92-R94) for reference.

I added comments in the tests for future readers in 97018fa59